R/gradethis_exercise_checker.R

```diff
@@ -271,6 +271,22 @@ prepare_check_env <- function(learnr_args, envir_caller = rlang::caller_env()) {
   }
 
   # Delayed evaluation of `.solution`
+  engine <- knitr_engine_caption(learnr_args[["engine"]])
+  if (!identical(engine, "R")) {
```

Comment:
Same here, I'd rather have a separate function or pattern to check for the exercise/code language, but I think that's a larger amount of work we'll want to tackle later on

Reply:
Quick question: why does `engine == "R"`, but `learnr_args$engine == "r"`?

Reply:
yeah that's totally one of the reasons to standardize. `knitr_engine_caption()` turns engine strings like `r`, `sql`, `javascript` into text suitable for the exercise caption, e.g. `R`, `SQL` or `JavaScript`. So `"r"` or `NULL` comes out as `"R"`.

Reply:
(In this case we're prepping the string for use in the message, so it makes sense to compare with `"R"`.)